// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.helloworld";
option java_outer_classname = "HelloWorldProto";
option objc_class_prefix = "HLW";

package helloworld;

// The greeting service definition.
service StaffManagement {
  // Sends a greeting
  // rpc LotteryGenerator (LotteryRequest) returns (LotteryResponse) {}
  rpc addDoctorProfile (DoctorDetials) returns (StorageReponse) {}
}

//for the lottery function 
// message LotteryRequest{
//   int32 randomNumber =1;
// }
// //reponse is yes/no
// message LotteryResponse{
//   string response=1;
// }


//to enter details of doctor and get response from mciroservices related to the addition of the identity
message DoctorDetials{
  string DoctorID=1;
  string Password=2;
  string Qualification=3;
  int32 DeptID=4;
  int32 Fees=5;
  string JobTitle=6;
}

message StorageReponse{ //response from the storage
  string message=1;  //returns about {sucess or error in this string}
}